tests/load/pipeline/test_model_item_format.py

```diff
@@ -0,0 +1,313 @@
+        primary_key="a",
+        table_name="example_table_1",
+        write_disposition=write_disposition,
+    )
+    pipeline.run(
+        [{"a": i} for i in range(10)],
+        primary_key="a",
+        table_name="example_table_2",
+        write_disposition=write_disposition,
+    )
+
+    # we now run a select of items 3-10 from example_table_2 into example_table_1
+    # each w_d should have a different outcome
+    dataset = pipeline.dataset()
+    example_table_columns = dataset.schema.tables["example_table_1"]["columns"]
+    # In Databricks, Ibis adds a helper column to emulate offset, causing a schema mismatch
+    # when the query attempts to insert it. We explicitly select only the expected columns.
+    relation = (
+        dataset["example_table_2"].order_by("a").limit(7, offset=3)[example_table_columns.keys()]
+    )
+    query = relation.query()
+
+    select_dialect = pipeline.destination.capabilities().sqlglot_dialect
+
+    @dlt.resource(
+        write_disposition=write_disposition, table_name="example_table_1", primary_key="a"
+    )
+    def copied_table() -> Any:
+        yield dlt.mark.with_hints(
+            SqlModel.from_sqlglot(query=query, dialect=select_dialect),
+            hints=make_hints(columns=example_table_columns),
+        )
+
+    pipeline.run([copied_table()])
+
+    # Snowflake is case sensitive
+    if destination_config.destination_type == "snowflake":
+        result_items = dataset["example_table_1"].df()["A"].tolist()
+    else:
+        result_items = dataset["example_table_1"].df()["a"].tolist()
+    result_items.sort()
+
+    if write_disposition == "merge":
+        # tables merged
+        assert result_items == list(range(10))
+    elif write_disposition == "replace":
+        # table fully replaced
+        assert result_items == [3, 4, 5, 6, 7, 8, 9]
+    elif write_disposition == "append":
+        # the middle part is duplicated
+        assert result_items == [0, 1, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 8, 9]
+    else:
+        raise ValueError(f"Unknown write disposition: {write_disposition}")
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        exclude=["athena", "dremio"],
+    ),
+    ids=lambda x: x.name,
+)
+def test_insert_less_or_reversed_columns(destination_config: DestinationTestConfiguration) -> None:
+    # NOTE: at least for duckdb, the column count AND order of the select query must match
+    # the target table schema
+    pipeline = destination_config.setup_pipeline("test_insert_less_columns", dev_mode=False)
+
+    pipeline.run(
+        [{"a": 1, "b": "n", "c": True}, {"a": None, "b": "n", "c": False}],
+        table_name="example_table",
+    )
+    dataset = pipeline.dataset()
+
+    example_table_columns = dataset.schema.tables["example_table"]["columns"]
+
+    select_dialect = pipeline.destination.capabilities().sqlglot_dialect
+
+    # This should raise a column number mismatch error
```

Comment:
If you are testing specific errors, it might make sense to inspect which exception exactly is raised and nested within PipelineStepFailed, so we are not shadowing this with some other problem.